server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -550,6 +551,35 @@ public Iterator<Setting<?>> settings() {
         Property.Final
     );
 
+    /**
+     * Legacy index setting, kept for 7.x BWC compatibility. This setting has no effect in 8.x. Do not use.
+     * TODO: Remove in 9.0
+     */
+    @Deprecated
+    public static final Setting<Integer> MAX_ADJACENCY_MATRIX_FILTERS_SETTING = Setting.intSetting(
+        "index.max_adjacency_matrix_filters",
+        100,
+        2,
+        new Setting.Validator<>() {
+            @Override
+            public void validate(Integer value) {}
+
+            @Override
+            public void validate(Integer value, Map<Setting<?>, Object> settings) {
+                validateVersionDependentDeprecatedSetting(MAX_ADJACENCY_MATRIX_FILTERS_SETTING.getKey(), settings);
+            }
+
+            @Override
+            public Iterator<Setting<?>> settings() {
+                final List<Setting<?>> settings = List.of(IndexMetadata.SETTING_INDEX_VERSION_CREATED);
+                return settings.iterator();
+            }
+        },
```

Comment:
Hmm I think we should do this validation automatically within the settings framework if a setting is marked as `Property.DeprecatedAndRemovedInCurrentMajor`. If we require implementors to remember to include this validator too then someone will forget and that'd be a bug.

Reply:
Fair enough, I can add the validation at lower level.